subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_common.c

```diff
@@ -155,3 +169,60 @@ int nrf_cloud_pending_fota_job_process(struct nrf_cloud_settings_fota_job * cons
 
 	return 0;
 }
+
+#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+int nrf_cloud_fota_fmfu_dev_set(const struct dfu_target_fmfu_fdev *const fmfu_dev_inf)
+{
+	if (!fmfu_dev_inf) {
+		return -EINVAL;
+	}
+
+	int ret;
+	const struct dfu_target_full_modem_params params = {
+		.buf = fmfu_buf,
+		.len = sizeof(fmfu_buf),
+		.dev = (struct dfu_target_fmfu_fdev *)fmfu_dev_inf
+	};
+
+	ret = dfu_target_full_modem_cfg(&params);
+	if (ret) {
+		LOG_ERR("Failed to initialize full modem FOTA: %d", ret);
+	} else {
+		fmfu_dev = *fmfu_dev_inf;
+	}
+
+	return ret;
+}
+
+int nrf_cloud_fota_fmfu_apply(void)
+{
+	int err;
+
+	err = nrf_modem_lib_shutdown();
+	if (err != 0) {
+		LOG_ERR("nrf_modem_lib_shutdown() failed: %d", err);
+		return err;
+	}
+
+	err = nrf_modem_lib_init(FULL_DFU_MODE);
+	if (err != 0) {
+		LOG_ERR("nrf_modem_lib_init(FULL_DFU_MODE) failed: %d", err);
+		(void)nrf_modem_lib_init(NORMAL_MODE);
+		return err;
+	}
+
+	err = fmfu_fdev_load(fmfu_buf, sizeof(fmfu_buf), fmfu_dev.dev, 0);
+	if (err != 0) {
+		LOG_ERR("Failed to apply full modem update, error: %d", err);
+		(void)nrf_modem_lib_init(NORMAL_MODE);
+		return err;
+	}
+
+	err = nrf_modem_lib_shutdown();
+	if (err != 0) {
+		LOG_WRN("nrf_modem_lib_shutdown() failed: %d\n", err);
+	}
+
```

Comment:
The nrf9160/http_update/full_modem_update/src/main.c file does all of these things, plus calls this at the end:
`	err = nrf_modem_lib_init(NORMAL_MODE);`
Is that not needed in your implementation?

Reply:
it is not needed if the application does a reboot.
TBD on how nrf_cloud_fota should handle this.  Add a parameter to the function?  Just leave it up to the application?  Open to suggestions.

Reply:
Note that there are recent commits related to TFM which affect when and how an unsecured application can reset the 9160.  It may require a specific Kconfig to allow it.